### PR TITLE
Vi/cleanup 

### PR DESCRIPTION
**Pull request by vincentvis**
Just a big cleanup and refactor of testing imports

Removes a whole bunch of import statements at the top and just imports `AllTestingModule` which contains everything (so we only have to add stuff for testing in one spot!)
This also makes the `Test.createTestingModule` into only importing one thing!

Another "big" change:
Changed the typeorm config!
Removed the conditional part which uses the .env.test file
now just uses a different class in the typeorm config for testing than for app.module.ts...

### CODE REVIEW

File: backend/src/auth/auth.spec.ts [hunk 1]
```diff
@@ -2,97 +2,29 @@ import { Test, TestingModule } from '@nestjs/testing';
...
-			name: 'u2',
-			password: 'p2',
-			password_confirm: 'p2',
-		},
-		{
-			name: 'u3',
-			password: 'p3',
-			password_confirm: 'p3',
-		},
-	];
```
Comment (Ryno95) [inline backend/src/auth/auth.spec.ts:50]: The mocking was removed cause it wasn't used for anything?

File: backend/src/pong/game/game.service.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { Injectable } from '@nestjs/common';
 import { InjectRepository } from '@nestjs/typeorm';
 import { AbstractService } from '../../shared/abstract.service';
-import { Repository } from 'typeorm';
+import { InsertResult, Repository } from 'typeorm';
```
Comment (Ryno95) [inline backend/src/pong/game/game.service.ts:4]: This import sneaked back in hehe

File: backend/src/users/user/user.controller.ts [hunk 3]
```diff
@@ -73,6 +73,13 @@ export class UserController {
 		});
 	}
 
+	// @UseGuards(AuthGuard)
+	// // localhost:8080/api/users/id/games
+	// @Get(':id/games')
+	// async findOneWithGame(@Param('id') id: number): Promise<Game[]> {
+	// 	return this.userService.getGames(id);
+	// }
+
```
Comment (Ryno95) [inline backend/src/users/user/user.controller.ts:82]: This sneaked back in as well, maybe with the pull from the main will resolve the last things changed.

**Review by Ryno95 — APPROVED** [review]
Ooooh so niiice! :D

File: backend/src/auth/auth.spec.ts (hunk 1 above)
Comment (vincentvis) [reply backend/src/auth/auth.spec.ts:50]: indeed

File: backend/src/pong/game/game.service.ts (hunk 2 above)
Comment (vincentvis) [reply backend/src/pong/game/game.service.ts:4]: pffffff
this is a "vscode is too slow" thing for me.. I open the file, it's all the same colour, I close the file....
Now I opened the file, waited 3 seconds, and the `InsertResult` became a different colour......

File: backend/src/users/user/user.controller.ts (hunk 3 above)
Comment (vincentvis) [reply backend/src/users/user/user.controller.ts:82]: this is probably just me sloppily merging main into this branch and picking the wrong option...

**Pull request merged**